code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -3608,6 +3608,26 @@
 	color = "#D9C0E7" //rgb: 217, 192, 231
 	custom_metabolism = 0.1
 
+/datum/reagent/ectoplasm	//This chem also slows external bleeding by 30% and internal bleeding by 70%
+	name = "Ectoplasm"
+	id = ECTOPLASM
+	description = "Concentrated ghostly essence."
+	reagent_state = LIQUID
+	color = "#02E816" //rgb: 2, 232, 22 (does not support alpha channels - yet!)
+	alpha = 140
+
+/datum/reagent/ectoplasm/on_mob_life(var/mob/living/M)
+
+	if(..())
+		return 1
+
+	M.adjustOxyLoss(-2 * REM)
+
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		H.shock_stage--
+		H.traumatic_shock--
```

Comment:
this actually does nothing because of the way pain works (or doesn't work, rather)

honk

you need to add it to /mob/living/carbon/proc/updateshock()